infra/tools/pr_helper/helper.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Script for implement PR helper."""
+import os
+import base64
+import requests
+import yaml
+
+OWNER = 'google'
+REPO = 'oss-fuzz'
+BASE_URL = f'https://api.github.com/repos/{OWNER}/{REPO}'
+
+def get_project_path(pr_number, headers):
+  """Gets the current project path."""
+  response = requests.get(f'{BASE_URL}/pulls/{pr_number}/files',
+                          headers=headers)
+  file = response.json()[0]
+  file_path = file['filename']
+  project_path = os.path.dirname(file_path)
+  return project_path
+
+
+def get_author_email(pr_number, headers):
+  """Retrieves the author's email address for a pull request, 
+  including non-public emails."""
+  commits_response = requests.get(f'{BASE_URL}/pulls/{pr_number}/commits',
+                                  headers=headers)
+  email = commits_response.json()[0]['commit']['author']['email']
+  return email
+
+
+def get_project_yaml(project_path, headers):
+  """Gets the project yaml file."""
+  project_response = requests.get(
+      f'{BASE_URL}/contents/{project_path}/project.yaml', headers=headers)
+  if project_response.status_code != 200:
+    return None
+
+  content = base64.b64decode(
+      project_response.json()['content']).decode('UTF-8')
+  return yaml.safe_load(content)
+
+
+def has_author_modified_project(project_path, pr_author, headers):
+  """Checks if the author has modified this project before."""
+  commits_response = requests.get(
+      f'{BASE_URL}/commits?path={project_path}&author={pr_author}',
+      headers=headers)
+
+  if commits_response.status_code != 200:
+    return False
+
+  commit = commits_response.json()[0]
+  return True, commit['sha']
+
+
+def get_pulls_url(commit, headers):
+  """Gets the pull request url."""
+  pr_response = requests.get(
+      f'{BASE_URL}/commits/{commit}/pulls', headers=headers)
+  if pr_response.status_code != 200:
+    return None
+  return pr_response.json()[0]['html_url']
+
+
+def save_output(message, is_ready_for_merge):
+  """Saves the github output."""
+  with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
+    print(f'message={message}', file=fh)
```

Comment:
Hmm...I guess this works, but could you just create a string and write it to the file using `fh.write`.
There's nothing wrong with what you have here, but I haven't seen this in our code before and would rather keep things consistent.